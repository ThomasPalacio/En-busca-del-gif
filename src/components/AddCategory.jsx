
import PropTypes  from 'prop-types';
import React, { useState } from 'react';



export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) =>{
        const {value} = target;
        setInputValue(value)
    }

    const onSubmit = (e) =>{
        e.preventDefault(); 
        const newInputValue = inputValue.trim();

        if (newInputValue.length <= 1) return;

        onNewCategory(newInputValue);
        setInputValue('');
    }

    
    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input 
                type='text'
                placeholder='Buscar Gif'
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
