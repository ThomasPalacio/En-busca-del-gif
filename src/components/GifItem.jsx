import PropTypes  from "prop-types"



export const GifItem = ({title,url}) => {
  return (
    <div className="card">
        <img src={url} alt={title}  className="card-img-top"/>
        <div className="card-body d-flex justify-content-center align-items-center">
          <p className="card-title">{title}</p>
        </div>
    </div>
    )
}

GifItem.propTypes = {
  title:PropTypes.string.isRequired,
  url:PropTypes.string.isRequired
}