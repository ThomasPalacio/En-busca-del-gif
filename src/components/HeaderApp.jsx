

export const HeaderApp = () => {
  return (
    <div className="d-flex justify-content-center">
        <div>
          <h1 className="textTitles">Buscador de GIF'S </h1>
          <h4 className="subtitleHeader">Descubr√≠ tu favorito en un instante.</h4>
        </div>
    </div>

  )
}
