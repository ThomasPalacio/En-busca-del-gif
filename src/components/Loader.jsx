
import { Dimmer, Loader, Image, Segment } from 'semantic-ui-react'

export const LoaderFunction = () => (
  <div className='container d-flex justify-content-center align-items-center'>
    <Segment>
      {/* <Dimmer active>
        <Loader>Loading</Loader>
      </Dimmer> */}

      <Image src='https://react.semantic-ui.com/images/wireframe/short-paragraph.png' />
    </Segment>
  </div>
)
