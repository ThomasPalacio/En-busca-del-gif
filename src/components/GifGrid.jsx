import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs";
import { LoaderFunction } from "./Loader";
import PropTypes  from "prop-types";



export const GifGrid = ({category}) => {
const categoryUpper = category.toUpperCase();
const {images,isLoading} = useFetchGifs(category);


  return (
    <div className="mt-5">
      
      <h2 className="textTitles d-flex align-item-center"><i className="fi fi-ss-bullet"></i>{categoryUpper}</h2>
      {
        isLoading && (<LoaderFunction/>)
      }
      
      <div className="card-grid">
      {images.map(({title,url,id}) => 
        <GifItem key={id} 
        title={title}
        url={url}
        />
      )}
      </div>
    </div>
  )
}

GifGrid.propTypes = {
  category:  PropTypes.string.isRequired,
}