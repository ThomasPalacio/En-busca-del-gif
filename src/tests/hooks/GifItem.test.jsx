import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe('Pruebas en <GifItem/>', () => {
    
    const title = 'Homero';
    const url = 'https://los-simpsons.com/homer.png'

    test('Debe hacer match con el snapshot', () => {
        
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    });


    test('Debe de mostrar la imagen con el url y el alt indicado', () => {
        
        render(<GifItem title={title} url={url}/>)
        // screen.debug()
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url)
        expect(alt).toBe(title);
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
    });
});