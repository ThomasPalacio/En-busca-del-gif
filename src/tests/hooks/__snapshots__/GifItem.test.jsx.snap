// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Homero"
      class="card-img-top"
      src="https://los-simpsons.com/homer.png"
    />
    <div
      class="card-body d-flex justify-content-center align-items-center"
    >
      <p
        class="card-title"
      >
        Homero
      </p>
    </div>
  </div>
</div>
`;
